api/gateway/middleware.go

```diff
@@ -34,3 +37,13 @@ func checkPostDisabled(state state.Module) mux.MiddlewareFunc {
 		})
 	}
 }
+
+// wrapRequestContext ensures we implement a deadline on serving requests
+// via the gateway server-side to prevent context leaks.
+func wrapRequestContext(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx, cancel := context.WithTimeout(r.Context(), time.Minute)
```

Comment:
N: extract the timeout in a separate func

Reply:
ok pls approve now :(